In [6]:
import pandas as pd
df = pd.read_csv(r"C:\Users\Huaweı\Downloads\Burak_restoran_yorumlar_ozetli.csv")
df.head()

Restoran  \
0               01_adana_durumluk   
1  01_adanali_ismail_ustanin_yeri   
2               06_ankara_kokorec   
3               100_more_alsancak   
4            10_numara_restaurant   

                                               Yorum  \
0  Ciğer şiş denedim çok beğendim. Etimesgutta sı...   
1  Mezeler etler tavuk harika\nHizmet süper\nHari...   
2  Türkiye'nin en iyi kokoreçlerinden biri 1 haft...   
3  Sushileri gerçekten cok basarili izmir roll fa...   
4  Berbat ötesi bir işletme. İstediğiniz yiyecekl...   

                                                Özet  
0  Restoran, özellikle Adana dürüm ve kebaplarıyl...  
1  Bu restoran, özellikle **lezzetli Adana kebabı...  
2  Bu restoran, özellikle kokoreç konusunda öne ç...  
3  %100&MORE Alsancak, İzmir'in popüler mekanları...  
4  Restoran, samimi ve sıcak bir ortamıyla öne çı...

In [8]:
# bert_similarity.py
from sentence_transformers import SentenceTransformer
from sklearn.metrics.pairwise import cosine_similarity
import numpy as np

In [9]:
def calculate_bert_similarity_per_row(comments, summaries):
    model = SentenceTransformer('dbmdz/bert-base-turkish-cased')
    similarities = []
    
    for i in range(len(comments)):
        # Her raw için ayrı ayrı hesapla
        comment = comments[i]
        summary = summaries[i]
        
        # Embedding'leri hesapla
        comment_embedding = model.encode([comment])
        summary_embedding = model.encode([summary])
        
        # Cosine similarity hesapla
        similarity = cosine_similarity(comment_embedding, summary_embedding)[0][0]
        similarities.append(similarity)
    
    return similarities

# Kullanım
similarities = calculate_bert_similarity_per_row(df['Yorum'], df['Özet'])
print(f"Her raw için benzerlik skorları: {similarities}")

No sentence-transformers model found with name dbmdz/bert-base-turkish-cased. Creating a new one with mean pooling.
c:\Users\Huaweı\AppData\Local\Programs\Python\Python312\Lib\site-packages\torch\nn\modules\module.py:1762: FutureWarning: `encoder_attention_mask` is deprecated and will be removed in version 4.55.0 for `BertSdpaSelfAttention.forward`.
  return forward_call(*args, **kwargs)
c:\Users\Huaweı\AppData\Local\Programs\Python\Python312\Lib\site-packages\torch\nn\modules\module.py:1762: FutureWarning: `encoder_attention_mask` is deprecated and will be removed in version 4.55.0 for `BertSdpaSelfAttention.forward`.
  return forward_call(*args, **kwargs)
c:\Users\Huaweı\AppData\Local\Programs\Python\Python312\Lib\site-packages\torch\nn\modules\module.py:1762: FutureWarning: `encoder_attention_mask` is deprecated and will be removed in version 4.55.0 for `BertSdpaSelfAttention.forward`.
  return forward_call(*args, **kwargs)
c:\Users\Huaweı\AppData\Local\Programs\Python\Python312\Lib

Her raw için benzerlik skorları: [np.float32(0.8997377), np.float32(0.9151245), np.float32(0.9036019), np.float32(0.88554347), np.float32(0.9356692), np.float32(0.89138275), np.float32(0.9110719), np.float32(0.92690784), np.float32(0.9096472), np.float32(0.92632747), np.float32(0.9103004), np.float32(0.9133166), np.float32(0.9111122), np.float32(0.8807708), np.float32(0.9029701), np.float32(0.82961434), np.float32(0.92926157), np.float32(0.77134216), np.float32(0.92606914), np.float32(0.9068335), np.float32(0.92476803), np.float32(0.92488), np.float32(0.8977268), np.float32(0.9100742), np.float32(0.9258423), np.float32(0.91901284), np.float32(0.90276736), np.float32(0.9270606), np.float32(0.9040622), np.float32(0.89196205), np.float32(0.9109889), np.float32(0.933898), np.float32(0.8928828), np.float32(0.92068464), np.float32(0.91540873), np.float32(0.88394696), np.float32(0.9409102), np.float32(0.92853314), np.float32(0.87741435), np.float32(0.92097193), np.float32(0.91467845), np.floa

In [ ]:
# Benzerlik skorlarını DataFrame'e çevir
# Restoran isimleri ile DataFrame oluştur
similarity_df = pd.DataFrame({
    'Restoran_Ismi': df['Restoran'],  # Restoran isimleri
    'Benzerlik_Skoru': similarities
})

# İlk 5 satırı göster
print("=== İLK 5 BENZERLİK SKORU ===")
print(similarity_df.head())

# Son 5 satırı da göster
print("\n=== SON 5 BENZERLİK SKORU ===")
print(similarity_df.tail())

# En yüksek 5 benzerlik skoru
print("\n=== EN YÜKSEK 5 BENZERLİK SKORU ===")
print(similarity_df.nlargest(5, 'Benzerlik_Skoru'))

# En düşük 5 benzerlik skoru
print("\n=== EN DÜŞÜK 5 BENZERLİK SKORU ===")
print(similarity_df.nsmallest(5, 'Benzerlik_Skoru'))

=== İLK 5 BENZERLİK SKORU ===
                    Restoran_Ismi  Benzerlik_Skoru
0               01_adana_durumluk         0.899738
1  01_adanali_ismail_ustanin_yeri         0.915124
2               06_ankara_kokorec         0.903602
3               100_more_alsancak         0.885543
4            10_numara_restaurant         0.935669

=== SON 5 BENZERLİK SKORU ===
                       Restoran_Ismi  Benzerlik_Skoru
2415     ziyafet_restaurant_catering         0.909345
2416  ziyafet_toplu_yemek_san_tic_as         0.735318
2417                     zoyas_house         0.906154
2418       zubeyir_ocakbasi_atasehir         0.921187
2419          zumrut_aile_restaurant         0.850218

=== EN YÜKSEK 5 BENZERLİK SKORU ===
            Restoran_Ismi  Benzerlik_Skoru
63         affan_ocakbasi         0.965965
2290         unluer_gurme         0.964223
349    balkanika_restoran         0.963604
626     cantine_del_frida         0.957928
581   cafe_italiano_sisli         0.956953

=== EN DÜŞÜK 

In [10]:
# Overall benzerlik hesaplama
def calculate_overall_similarity(similarities):
    """
    Ayrı ayrı benzerlik skorlarından overall istatistikler hesapla
    """
    overall_stats = {
        'mean_similarity': np.mean(similarities),
        'median_similarity': np.median(similarities),
        'std_similarity': np.std(similarities),
        'min_similarity': np.min(similarities),
        'max_similarity': np.max(similarities),
        'total_restaurants': len(similarities)
    }
    
    return overall_stats

# Overall benzerlik hesapla
overall_stats = calculate_overall_similarity(similarities)

print("=== OVERALL BENZERLİK İSTATİSTİKLERİ ===")
print(f"Toplam Restoran Sayısı: {overall_stats['total_restaurants']}")
print(f"Ortalama Benzerlik: {overall_stats['mean_similarity']:.4f}")
print(f"Medyan Benzerlik: {overall_stats['median_similarity']:.4f}")
print(f"Standart Sapma: {overall_stats['std_similarity']:.4f}")
print(f"Minimum Benzerlik: {overall_stats['min_similarity']:.4f}")
print(f"Maksimum Benzerlik: {overall_stats['max_similarity']:.4f}")

# Benzerlik dağılımı
print(f"\nBenzerlik Dağılımı:")
print(f"0.0-0.2: {sum(1 for x in similarities if 0.0 <= x < 0.2)} restoran")
print(f"0.2-0.4: {sum(1 for x in similarities if 0.2 <= x < 0.4)} restoran")
print(f"0.4-0.6: {sum(1 for x in similarities if 0.4 <= x < 0.6)} restoran")
print(f"0.6-0.8: {sum(1 for x in similarities if 0.6 <= x < 0.8)} restoran")
print(f"0.8-1.0: {sum(1 for x in similarities if 0.8 <= x <= 1.0)} restoran")

=== OVERALL BENZERLİK İSTATİSTİKLERİ ===
Toplam Restoran Sayısı: 2420
Ortalama Benzerlik: 0.9025
Medyan Benzerlik: 0.9072
Standart Sapma: 0.0316
Minimum Benzerlik: 0.6556
Maksimum Benzerlik: 0.9660

Benzerlik Dağılımı:
0.0-0.2: 0 restoran
0.2-0.4: 0 restoran
0.4-0.6: 0 restoran
0.6-0.8: 53 restoran
0.8-1.0: 2367 restoran


0.0-0.2: 0 restoran    → 0 restoran çok düşük benzerlik
0.2-0.4: 0 restoran    → 0 restoran düşük benzerlik
0.4-0.6: 0 restoran    → 0 restoran orta benzerlik
0.6-0.8: 53 restoran    → 53 restoran yüksek benzerlik
0.8-1.0: 2367 restoran    → 2367 restoran çok yüksek benzerlik

Cosine Similarity değerleri:
1.0 = Tamamen aynı (100% benzer)
0.0 = Hiç benzer değil (0% benzer)
0.5 = Orta düzey benzerlik (50% benzer)